api/src/main/java/org/pmiops/workbench/auth/ProfileService.java

```diff
@@ -70,7 +75,32 @@ public Profile getProfile(User user) throws ApiException {
     if (user.getAuthorities() != null) {
       profile.setAuthorities(new ArrayList(user.getAuthorities()));
     }
-
+    String userEmailVerificationStatus = null;
+    if (user.getEmailVerificationStatus().equals(EmailVerificationStatus.UNVERIFIED)) {
+      profile.setEmailVerificationStatus(EmailVerificationStatus.UNVERIFIED);
+    } else if (user.getEmailVerificationStatus().equals(EmailVerificationStatus.VERIFIED)) {
+      profile.setEmailVerificationStatus(EmailVerificationStatus.VERIFIED);
+    } else {
+      // Verification is pending, need to query mailchimp.
+      try {
+        userEmailVerificationStatus = mailChimpService.getMember(user.getContactEmail());
+      } catch (NotFoundException e) {
+        profile.setEmailVerificationStatus(EmailVerificationStatus.UNVERIFIED);
+        user.setEmailVerificationStatus(EmailVerificationStatus.UNVERIFIED);
+        userDao.save(user);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      if (userEmailVerificationStatus != null) {
+        if (userEmailVerificationStatus.equals(MailChimpService.MAILCHIMP_PENDING)) {
+          profile.setEmailVerificationStatus(EmailVerificationStatus.PENDING);
+        } else {
```

Comment:
This else covers too many cases -- there are other statuses described here:

https://developer.mailchimp.com/documentation/mailchimp/guides/manage-subscribers-with-the-mailchimp-api/

In particular, there's unsubscribed and cleaned. We probably need to handle those somehow, particularly cleaned (as somebody can enter in a bad e-mail address.)